### ISSUE TITLE
Rule proposal: `no-export-function-then`

### ISSUE THREAD

**Issue by fregante** [sindresorhus/eslint-plugin-unicorn#1651]
### Description

Kind of a niche rules since it's rare both in usage and in pitfall. 

Context: `await` works on Promise-like objects and `import()` creates an object with every `export` of the specified module.

```js
// ./my-module.js
export function then () {
	throw new Error('You shouldn’t have called me')
}
```

```js
await import('./my-module');
// Error: You shouldn’t have called me
```

### Fail

```js
export function then() {}
export const then = () => {}

const then = () => {};
export {then};
```

### Pass

```js
export function therefore() {}
export const notThen = () => {}

const totallyNotThen = () => {};
export {totallyNotThen};

export const then = 1999; // 😬 
```

**Comment by fisker** [comment]
#1616?

**Comment by fregante** [comment]
Exactamundo

**Comment by fisker** [comment]
Stole  your examples in https://github.com/sindresorhus/eslint-plugin-unicorn/pull/1616/commits/e9a937c5a03b074c19031603370d5c5112262716
